pkg/registry/batch/job/strategy.go

```diff
@@ -237,8 +251,9 @@ func generateSelector(obj *batch.Job) {
 	if obj.Spec.Selector.MatchLabels == nil {
 		obj.Spec.Selector.MatchLabels = make(map[string]string)
 	}
-	if _, found := obj.Spec.Selector.MatchLabels["controller-uid"]; !found {
-		obj.Spec.Selector.MatchLabels["controller-uid"] = string(obj.ObjectMeta.UID)
+
+	if _, found := obj.Spec.Selector.MatchLabels[batch.ControllerUidLabel]; !found {
```

Comment:
just to be sure I follow - existing Jobs, which have the "naked" label also have a selector which selects that label, so there's no risk of abandonment here, right?  New instances of Job will have the new label and the new selector, IIUC?

Reply:
From what @alculquicondor and I discussed I believe this is the case.  

We don't change selector or labels on updates.

Reply:
That is correct, existing jobs are unaffected. New jobs will have the new selector.